dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/guestbook/GuestbookPage.java

```diff
@@ -238,45 +240,22 @@ public String save() {
             }
         }
 
-        Command<Dataverse> cmd;
-        try {
-            if (editMode == EditMode.CREATE || editMode == EditMode.CLONE) {
-                guestbook.setCreateTime(new Timestamp(new Date().getTime()));
-                guestbook.setUsageCount(new Long(0));
-                guestbook.setEnabled(true);
-                dataverse.getGuestbooks().add(guestbook);
-                cmd = new UpdateDataverseCommand(dataverse, null, null, dvRequestService.getDataverseRequest(), null);
-                commandEngine.submit(cmd);
-                create = true;
-            } else {
-                cmd = new UpdateDataverseGuestbookCommand(dataverse, guestbook, dvRequestService.getDataverseRequest());
-                commandEngine.submit(cmd);
-            }
+        if (editMode == EditMode.CREATE || editMode == EditMode.CLONE) {
+            Try<Dataverse> guestbookTry = Try.of(() -> guestbookService.saveGuestbook(guestbook))
+                    .onSuccess(dv -> JsfHelper.addFlashMessage(BundleUtil.getStringFromBundle("guestbook.create")))
+                    .onFailure(this::handleErrorMessages);
 
-        } catch (EJBException ex) {
-            StringBuilder error = new StringBuilder();
-            error.append(ex).append(" ");
-            error.append(ex.getMessage()).append(" ");
-            Throwable cause = ex;
-            while (cause.getCause() != null) {
-                cause = cause.getCause();
-                error.append(cause).append(" ");
-                error.append(cause.getMessage()).append(" ");
+            if(guestbookTry.isFailure() && guestbookTry.getCause() instanceof EJBException) {
```

Comment:
I wouldn't check if this is EJBException, since they didn't ever use ApplicationException annotation(so exception unrolling), so if there will be CommandException it will work incorrectly, I would check if there is a failure and that's it.

Reply:
You have, correctly, pointed out that the previous code wouldn't work the same. In the original code then only returned null for EJBException and did nothing specific for CommandException so this code should work the same as before.

Reply:
Ah yes, I forgot it was checked exception.